troi/patch.py

```diff
@@ -1,12 +1,33 @@
+import logging
 from abc import ABC, abstractmethod
 
 import troi
 
 
 class Patch(ABC):
 
-    def __init__(self):
-        pass
+    def __init__(self, debug=False):
+        if debug:
+            level = logging.DEBUG
+        else:
+            level = logging.INFO
+        logging.basicConfig(level=level)
+        self.logger = logging.getLogger(type(self).__name__)
```

Comment:
I was looking at the logging docs, loggers are hierarchical with . as a separator. Therefore there might be value in using `type(self)` as the logger. This would allow us to configure a logger `troi`, and have it apply to all child loggers. Theoretically this would allow us to do things like having different logging levels for elements and patches, although I'm not sure if this is something that we have a specific requirement for

Reply:
I don't see it for right now -- lets keep it simple and if we learn we need better debug support, we'll add it.